### PR TITLE
Include data only in dynamic libraries for Dart

### PR DESCRIPTION
**Pull request by mosuem**
Set `if (linkMode == LinkMode.dynamic) 'compiled_data',`, and some cosmetic changes.

### CODE REVIEW

File: ffi/dart/tool/build_libs.dart [hunk 1]
```diff
@@ -20,34 +20,45 @@ Future<void> main(List<String> args) async {
...
+      'rustup',
+      rustupArguments,
+      'Exited with a non-zero exit code',
+      rustupExitCode,
+    );
   }
 
-  final cargo = await Process.start('cargo', [
+  final features = [
+    if (linkMode == LinkMode.dynamic) 'compiled_data',
```
Comment (mosuem) [inline ffi/dart/tool/build_libs.dart:56]: The actual change

**Review by robertbastian — CHANGES_REQUESTED** [review]
Now that we have feature-passing logic, I'd rather have this be explicitly decided by the caller than derived from the linking type.

File: ffi/dart/tool/build_libs.dart [hunk 2]
```diff
@@ -14,40 +14,52 @@ Future<void> main(List<String> args) async {
...
+      rustupExitCode,
+    );
   }
 
-  final cargo = await Process.start('cargo', [
+  final features = [
+    'buffer_provider',
+    if (isNoStd) ...['libc-alloc', 'panic-handler'],
+    if (!isNoStd) ...['logging', 'simple_logger'],
+  ].join(',');
```
Comment (robertbastian) [inline ffi/dart/tool/build_libs.dart:61]: nit: spread the `cargoFeatures` in here

File: ffi/dart/tool/build_libs.dart [hunk 3]
```diff
@@ -68,7 +68,7 @@ Future<void> buildLib(
     '--config=profile.release.panic="abort"',
     '--config=profile.release.codegen-units=1',
     '--no-default-features',
-    '--features=$features,$cargoFeatures',
+    '--features=${[...features, ...cargoFeatures].join(',')}',
```
Comment (robertbastian) [inline ffi/dart/tool/build_libs.dart:71]: cargo features and features are the same thing, they should be merged earlier

**Review by robertbastian — APPROVED** [review]
Also please merge this into main

**Pull request merged**